plugins/kafka-backend/src/service/router.ts

```diff
@@ -68,9 +78,12 @@ export const makeRouter = (
         }));
       }),
     );
+
     res.json({ consumerId, offsets: groupWithTopicOffsets.flat() });
   });
 
+  router.use(errorHandler());
```

Comment:
Shouldn't add this here should we? Up to the backend to decide how errors are treated. Right now we register an error handler at the end in the service builder.

By converting to an error response at this point to don't let the backend register custom error handling that could otherwise potentially trace errors or whatnot. Is there any reason to add it within the plugin?

Reply:
Interesting point. Who should handle errors? Is not the error payload format etc strictly part of the individual plugin API contract? Consumers should be able to use the default client and talk to any catalog - right? I'm not sure that this should be under the control of the org.

Frankly what I did was to copy what the [catalog does](https://github.com/backstage/backstage/blob/master/plugins/catalog-backend/src/service/router.ts#L234), because it was convenient for the test (I needed to see it get converted to a 404 - not a 500) and made sense in my mind.

Reply:
Tracing errors could of course still be done in a wrapping middleware - but by observing the external behavior of the catalog (its HTTP responses), rather than internal implementation details. Which arguably is a good thing.

Reply:
We decided in the end that this is better left out after all :)